apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -313,26 +315,23 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
 
   defp retrieve_trigger_configuration(realm_name, trigger_id) do
     query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(
-        "SELECT value FROM kv_store WHERE group='triggers' AND key=:trigger_id;"
-      )
-      |> DatabaseQuery.put(:trigger_id, trigger_id)
-
-    with {:ok, client} <- Database.connect(realm: realm_name),
-         {:ok, result} <- DatabaseQuery.call(client, query),
-         [value: trigger_data] <- DatabaseResult.head(result),
-         trigger <- Trigger.decode(trigger_data),
+      from kvstore in KvStore,
+        prefix: ^realm_name,
```

Comment:
Since https://github.com/astarte-platform/astarte/pull/909, the Astarte realm name may differ from the  actual keyspace name in the database.
This was hidden in the `Database.connect` call, but now that it is being removed, the keyspace should be explicitly set. See e.g. here for the conversion: https://github.com/astarte-platform/astarte/blob/d5ed3c0a6c1af3b620a436ae506462c756537f6d/apps/astarte_trigger_engine/lib/astarte_trigger_engine/policy/queries.ex#L37-L38